docs/mdbook/src/android/geojson-guide.md

```diff
@@ -0,0 +1,349 @@
+
+A typical GeoJSON file might look like:
+```json
+ {
+  "type": "FeatureCollection",
+  "features": [
+    {
+      "type": "Feature",
+      "properties": {},
+      "geometry": {
+        "type": "Polygon",
+        "coordinates": [
+          [
+            [
+              -77.06867337226866,
+              38.90467655551809
+            ],
+            [
+              -77.06233263015747,
+              38.90479344272695
+            ],
+            [
+              -77.06234335899353,
+              38.906463238984344
+            ],
+            [
+              -77.06290125846863,
+              38.907206285691615
+            ],
+            [
+              -77.06364154815674,
+              38.90684728656818
+            ],
+            [
+              -77.06326603889465,
+              38.90637140121084
+            ],
+            [
+              -77.06321239471436,
+              38.905561553883246
+            ],
+            [
+              -77.0691454410553,
+              38.905436318935635
+            ],
+            [
+              -77.06912398338318,
+              38.90466820642439
+            ],
+            [
+              -77.06867337226866,
+              38.90467655551809
+            ]
+          ]
+        ]
+      }
+    }
+  ]
+}
+```
+So far we learned describing Geospatial data in GeoJSON files. We will start applying this knowledge into our map applications 
+
+### 4. GeoJsonSource
+As we discussed before, map requires some sort data to be rendered. We use different sources such as [`Vector`]("https://maplibre.org/maplibre-native/android/api/-map-libre%20-native%20-android/org.maplibre.android.style.sources/-vector-source/index.html?query=class%20VectorSource%20:%20Source"), [`Raster`]("https://maplibre.org/maplibre-native/android/api/-map-libre%20-native%20-android/org.maplibre.android.style.sources/-raster-source/index.html?query=class%20RasterSource%20:%20Source") and `GeoJSON`.
+We will focus exclusively on `GeoJsonSource` and will not address other sources.
+`GeoJsonSource` is a type of source that has a unique `String` ID and GeoJSON data.
+
+There are several ways to construct a `GeoJsonSource`:
+- Locally stored files such as assets and raw folders
+- Remote services 
+- Raw string  parsed into FeatureCollections objects
+- Geometry, Feature, and FeatureCollection objects that map to GeoJSON Base builders
+
+A sample GeoJsonSource:
+```kotlin
+val source = GeoJsonSource("source", featureCollection)
+val lineLayer = LineLayer("layer", "source")
+    .withProperties(
+        PropertyFactory.lineColor(Color.RED),
+        PropertyFactory.lineWidth(10f)
+    )
+
+style.addSource(source)
+style.addLayer(lineLayer)
+```
+
+Note that you can not simply show data on a map. Layers must reference them.
+Therefore, you create a layer that gives visual appearance to it.
+
+- Loading from local files
+
+with assets folder file
+```kotlin
```

Comment:
All code snippets should be referenced so that we can be sure they compile in the future.